java/src/main/native/src/CudfJni.cpp

```diff
@@ -468,6 +471,44 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Cudf_filter(JNIEnv *env, jclass, jlo
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Cudf_replaceNulls(JNIEnv *env, jclass, jlong input_jcol,
+                                                              jlong r_int_values, jfloat r_f_value,
+                                                              jdouble r_d_value,
+                                                              jboolean r_is_valid, int r_dtype) {
+  JNI_NULL_CHECK(env, input_jcol, "input column is null", 0);
+  try {
+    gdf_column *input = reinterpret_cast<gdf_column *>(input_jcol);
+    gdf_scalar replacement{};
+    cudf::jni::gdf_scalar_init(&replacement, r_int_values, r_f_value, r_d_value, r_is_valid,
+                               r_dtype);
+    std::unique_ptr<gdf_column, decltype(free) *> result(
```

Comment:
Did we decide on not wrapping this in gdf_column_wrapper? Apologize if we have gone over this already

Reply:
@razajafri thanks for bringing this up again, I had responded internally but we didn't discuss further. 

I think that while things are in flux (to cpp land) that we are mixing things like `new` and `free` in various places. `gdf_column_wrapper` for example new's up the `gdf_column` which we later `free` using `freeCudfColumn`. So I think this is bigger than this PR, but we need to address it. At the moment, the code here is not mixing things, so I prefer the current implementation. 

Additionally, once `class cudf::column` exists (https://github.com/rapidsai/cudf/pull/1995#discussion_r293620608), we should remove the wrapper entirely.

Reply:
Makes sense to make all the changes once when we have `class cudf::column`. Thanks